### PR TITLE
Add gitlab ops guide

### PR DESCRIPTION
**Pull request by michaellzc**

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3831** <a href="https://app.graphite.dev/github/pr/sourcegraph/handbook/3831" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/handbook/3831?utm_source=stack-comment).

File: content/departments/engineering/engineering/tools/infrastructure/gitlab.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# [gitlab.sgdev.org](https://gitlab.sgdev.org)
+
+## Overview
+
+We use [ArgoCD](../process/deployments/index.md#argocd) to sync the GitLab deployment by following GitOps practices. All configuration is tracked in [sourcegraph/infrastructure] repo.
```
Comment (michaellzc) [inline content/departments/engineering/engineering/tools/infrastructure/gitlab.md]:
```suggestion
We use [ArgoCD](../../process/deployments/index.md#argocd) to sync the GitLab deployment by following GitOps practices. All configuration is tracked in [sourcegraph/infrastructure] repo.
```

**Review by kalanchan — APPROVED** [review]

**Review by bobheadxi — APPROVED** [review]
sweeettttt

File: content/departments/engineering/dev/tools/infrastructure/gitlab.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+Create a Pull Request to make neccessary changes and ask for review. Once the PR is merged, [argocd] will sync the new changes to the cluster.
+
+### How to upgrade/change (manual)
+
+> Only use this approach when [ArgoCD] is down
+
+Switch context to `dogfood` cluster
+
+```sh
+kubectx gke_sourcegraph-dogfood_us-central1-f_dogfood
```
Comment (sanderginn) [inline content/departments/engineering/dev/tools/infrastructure/gitlab.md:43]:
```suggestion
kubectx gke_sourcegraph-dogfood_us-central1-f_dogfood

# or if you don't have kubectx installed
# also, you should check it out https://github.com/ahmetb/kubectx
kubectl config set current-context gke_sourcegraph-dogfood_us-central1-f_dogfood
```

**Review by sanderginn — APPROVED** [review]
👍 

Btw, the other day I found out about [kubectx aliases](https://github.com/ahmetb/kubectx#examples). Maybe you know about them already, just putting it out there because it's a nice feature

**Review by filiphaftek — APPROVED** [review]

**Review by davejrt — APPROVED** [review]

**Pull request merged**